Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/Notify.java

```diff
@@ -0,0 +1,43 @@
+/*-
+ * #%L
+ * This file is part of "Apromore Core".
+ * %%
+ * Copyright (C) 2018 - 2020 Apromore Pty Ltd.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.apromore.portal.common;
+
+import org.zkoss.zk.ui.util.Clients;
+
+/**
+ * Utilities to abstract notification
+ */
+public class Notify {
+
+    public static final void show(String message, String type) {
+        Clients.evalJavaScript("Ap.common.notify('" + message.replace("'", "\\'") + "','" + type + "');");
+    }
+
+    public static final void info(String message) {
+        Clients.evalJavaScript("Ap.common.notify('" + message.replace("'", "\\'") + "','info');");
+    }
+
+    public static final void error(String message) {
+        Clients.evalJavaScript("Ap.common.notify('" + message.replace("'", "\\'") + "','error');");
+    }
+
+}
```

Comment:
"Notify" is not a right name for a class as it's confusing with a method name, e.g. Ap.common.notify() is a method used in the javascript code. Suggest to change it to Notification, MessageBox, or a noun. Naming is a simple coding step but has a lasting effect on the code readability. It seems not a big deal here in this particular Notify.java file but I would suggest we try to stick to good coding common sense.

Reply:
Another comment: is there a good reason to change from using ZK's built-in facilities to our own implementation? I guess it's a yes of course and you can write your justification here.

Reply:
On the other hand, do we have a unit test for this new class? For example, the message could have special characters, HTML tags. What happens if type is an arbitrary value? The unit tests would be a good place to test and also put these rules as a kind of documentation for anybody to understand how to use this utility which seems to be a good one for us.

Reply:
A similar style of this contemporary notification is available only in ZK 9. This is just a class for abstracting the ZK implementation later on. Moreover, this supports HTML snippets like bold important text.

Reply:
If ZK 9 has feature like this we actually want, it might be worth our while to do the upgrade.  I'm certainly open to investigating how much effort it'd take.